### PR TITLE
Fixed binary numeric I/O methods and test cases

### PR DESCRIPTION
**Pull request by jhh67**
- Fixed test cases that incidentally relied on binary numeric conversions in formatted I/O.
- Updated output from tests cases that rely on binary string and type conversions to include
    deprecation warning as there aren't replacement methods for these yet.
- Removed default value for `ioendian` argument to `writeBinary` and `readBinary`.
- Fixed the documentation for `readBinary` and `writeBinary`.

### CODE REVIEW

File: test/exercises/Mandelbrot/MPlot.chpl [hunk 1]
```diff
@@ -160,21 +161,24 @@ const header_size = 14;
...
   // Write the BMP image header
-  outfile.writef("BM%<4u %<2u %<2u %<4u",
-                 size,
-                 0 /* reserved1 */,
-                 0 /* reserved2 */,
-                 offset_to_pixel_data);
-  
+  outfile.writef("BM");
+  outfile.writeBinary(size:uint(32), ioendian.little);
+  outfile.writeBinary(0:uint(16), ioendian.little); /* reserved1 */
```
Comment (lydia-duncan) [inline test/exercises/Mandelbrot/MPlot.chpl:166]: Does this lose the spaces between what you're writing?

File: test/exercises/c-ray/Image.chpl [hunk 2]
```diff
@@ -120,6 +120,29 @@ proc writeImageBMP(outfile, pixels) {
...
+  outfile.writeBinary(0:uint(32), ioendian.little); /* no compression */
+  outfile.writeBinary(pixelsSize:uint(32), ioendian.little);
+  outfile.writeBinary(2835:uint(32), ioendian.little); /*pixels/meter print resolution=72dpi*/
+  outfile.writeBinary(2835:uint(32), ioendian.little); /*pixels/meter print resolution=72dpi*/
+  outfile.writeBinary(0:uint(32), ioendian.little); /* colors in palette */
+  outfile.writeBinary(0:uint(32), ioendian.little); /* "important" colors */
+
+
+
+/*
```
Comment (lydia-duncan) [inline test/exercises/c-ray/Image.chpl:145]: I think the code following this can just be removed rather than commented out

**Review by lydia-duncan — COMMENTED** [review]
Does the `test/types/bytes/io/formatNonUTF8.good` file really need to be added?  Looking at your original PR, I wouldn't expect it to need a new .good file, so I'm wondering if this was generated and isn't appropriately .gitignored or something.

Otherwise, just one update that shouldn't be a major issue and one question that I'm surprised didn't turn up in testing (though maybe there's something I'm not understanding about binary IO and how it handles spaces)

**Comment by lydia-duncan** [comment]
I might just be misreading how Github represents binary files, so maybe ignore the question about the .good file

File: test/exercises/Mandelbrot/MPlot.chpl (hunk 1 above)
Comment (jhh67) [reply test/exercises/Mandelbrot/MPlot.chpl:166]: This is very interesting. I tried a sample program that wrote two binary integers using the format string `%|4i %|4i` and it didn't insert the space between them. I don't know if this is by design or a bug. The headers in MPlot.chpl don't have spaces so converting to `writeBinary` worked. But I need to investigate the space issue more.

Comment (mppf) [reply test/exercises/Mandelbrot/MPlot.chpl:166]: I think it was intentional but it is moot since we are deprecating it, right?

**Review by lydia-duncan — APPROVED** [review]
Awesome, thanks!

**Pull request merged**